src/volue/mesh/tests/test_model.py

```diff
@@ -0,0 +1,444 @@
+            child_object_full_info = session1.get_object(child_object_path, full_attribute_info=True)
+            assert child_object_full_info.name == 'TestPlantChild'
+
+    with connection.create_session() as session2:
+        
+            parent_object_path = f"{root_object_ownership_relation_attribute_parent}/{'TestPlantParent'}"
+    
+            parent_object_id = session2.get_object(parent_object_path).id
+            session2.delete_object(parent_object_id)
+            # commit
+            session2.commit()
+
+            # check that parent object is deleted
+            with pytest.raises(grpc.RpcError, match="Object not found"):
+                parent_object_full_info = session2.get_object(parent_object_path, full_attribute_info=True)
+                
+        
+@pytest.mark.database
+def test_two_sessions_adding_child_after_removing_parent(connection):
+        
+    root_object_ownership_relation_attribute_parent = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef"
+    root_object_ownership_relation_attribute_child = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef/TestPlantParent.SimpleOwnershipAtt"
+        
+    session1 = connection.create_session()
+    session1.open()
+  
+    session2 = connection.create_session()
+    session2.open()
+   
+    #first create the parent object
+    session1.create_object(root_object_ownership_relation_attribute_parent,"TestPlantParent")
+    session1.commit()
+    # create child object
+    session1.create_object(root_object_ownership_relation_attribute_child,"TestPlantChild")
+    
+    #delete parent object    
+    parent_object_path = f"{root_object_ownership_relation_attribute_parent}/{'TestPlantParent'}"
+    
+    parent_object_id = session2.get_object(parent_object_path).id
+    session2.delete_object(parent_object_id)
+    
+    # commit the deletion of parent
+    session2.commit()
+    
+    # commit on creation of child should fail
+    with pytest.raises(grpc.RpcError, match="Missing owner"):
+        session1.commit()  
+   
+    session1.close()
+    session2.close()
+    
+        
+@pytest.mark.database
+def test_two_sessions_adding_child_after_removing_parent2(connection):
+     
+    root_object_ownership_relation_attribute_parent = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef"
+    root_object_ownership_relation_attribute_child = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef/TestPlantParent.SimpleOwnershipAtt"
+        
+    session1 = connection.create_session()
+    session1.open()
+  
+    session2 = connection.create_session()
+    session2.open()
+   
+    #first create the parent objectd
+    session1.create_object(root_object_ownership_relation_attribute_parent,"TestPlantParent")
+    session1.commit()
+    
+    #delete parent object    
+    parent_object_path = f"{root_object_ownership_relation_attribute_parent}/{'TestPlantParent'}"
```

Comment:
Instead of manually creating a path to the newly created object you could use the return object from the `create_object` itself.